internal/testutility/utility.go

```diff
@@ -57,3 +57,18 @@ func ValueIfOnWindows(win, or string) string {
 
 	return or
 }
+
+// CreateTestDir makes a temporary directory for use in testing that involves
+// writing and reading files from disk - make sure to call the cleanup function!
+func CreateTestDir(t *testing.T) (string, func()) {
+	t.Helper()
```

Comment:
`t.Cleanup` would be helpful here.
https://pkg.go.dev/testing#T.Cleanup

no need to make caller remember to call cleanup function

Reply:
oh nice - and yeah I just realised `t.TempDir` is a thing though sadly it doesn't support a prefix which I personally like to easily identify and manually clean up in the case of a failure or debugging.

Oh well, still probably best to just switch to using that

Reply:
Sounds reasonable. At least `t.Cleanup` sounds useful in this case.